### PR TITLE
Implements thesis processing form

### PR DESCRIPTION
**Pull request by matt-bernhardt**
Details in the commit message. More verbose history in the ticket:

https://mitlibraries.atlassian.net/browse/ETD-309

The follow-up ticket for the publication status field is https://mitlibraries.atlassian.net/browse/ETD-343

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Review by JPrevost — CHANGES_REQUESTED** [review]
I'd prefer we update the `active_holds?` and `authors_graduated?` to return booleans to match the typical expectations in ruby applications. If you feel that leads to significantly more logic where the data is used in the application, let me know and we can discuss it further.

File: test/models/thesis_test.rb [hunk 1]
```diff
@@ -501,4 +501,57 @@ class ThesisTest < ActiveSupport::TestCase
...
+  end
+
+  test 'authors_graduated? returns "No" if sole author has not graduated' do
+    thesis = theses(:one)
+    assert_equal 1, thesis.authors.count
+    assert_equal false, thesis.authors.first.graduation_confirmed
+    assert_equal "No", thesis.authors_graduated?
+  end
+
+  test 'authors_graduated? returns "No" if any author has not graduated' do
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply test/models/thesis_test.rb:551]: I can add this test, sure

File: app/models/thesis.rb [hunk 2]
```diff
@@ -94,6 +96,19 @@ class Thesis < ApplicationRecord
     select { |t| VALID_MONTHS.include? t.grad_date.strftime('%B') }
   }
 
+  # Returns a true/false value (rendered as "yes" or "no") if there are any
+  # holds with a status of either 'active' or 'expired'. A false/"No" is
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/models/thesis.rb:100]: Yeah, I was on the fence about whether this method should just return true/false and leave the formatting of that response to the view layer - so thanks for the nudge in this direction. I'll make the change - it shouldn't be too hard to update the view accordingly.

File: app/views/thesis/_department_thesis_fields.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  <%= f.input :thesis_id, as: :hidden %>
+  <%= f.input :department_id, as: :select,
+                        collection: Department.order(:name_dw),
+                        selected: f.object.department_id,
+                        required: true,
+                        validate: { presence: true },
+                        label: 'Department *',
+                        label_method: :name_dw,
+                        input_html: { class: 'field field-select wide' } %>
+  <aside><%= link_to_remove_association "Remove this department", f %></aside>
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/views/thesis/_department_thesis_fields.html.erb:12]: This... yeah. There's a lot of potential complexity here, and I'm worried about overly complex technical implementations when what's needed is a user who doesn't do really illogical things.

Many, many years ago I worked on a UI widget that might be preferrable - http://jsfiddle.net/matt_bernhardt/22kUX/ - but I agree with at most making this a ticket to tackle in the future, if it proves needed.

[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/views/thesis/_department_thesis_fields.html.erb:12]: I've created ticket https://mitlibraries.atlassian.net/browse/ETD-350 in our backlog to record the need for this work.

**Comment by matt-bernhardt** [comment]
Ok, I think I've now responded to the three requested changes. Two of them have been addressed with code changes that should solve the issue, while the third has spawned a new ticket in our backlog.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**